src/sentry/models/scheduledeletion.py

```diff
@@ -94,13 +121,44 @@ def get_model(self):
     def get_instance(self):
         return self.get_model().objects.get(pk=self.object_id)
 
-    def get_actor(self):
-        from sentry.models import User
-
+    def get_actor(self) -> APIUser | None:
         if not self.actor_id:
             return None
 
-        try:
-            return User.objects.get(id=self.actor_id)
-        except User.DoesNotExist:
-            return None
+        return user_service.get_user(user_id=self.actor_id)
```

Comment:
I can validate this is safe -- I have tracked all `on_delete` call sites and they are compatible with `APIUser`.